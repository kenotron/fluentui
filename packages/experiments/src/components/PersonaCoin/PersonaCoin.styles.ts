import { IPersonaCoinComponent } from './PersonaCoin.types';
import { initialsColorPropToColorCode } from 'office-ui-fabric-react/lib/components/Persona/PersonaInitialsColor';

export const DEFAULT_PERSONA_COIN_SIZE = 48;

export const PersonaCoinStyles: IPersonaCoinComponent['styles'] = props => {
  const { size = DEFAULT_PERSONA_COIN_SIZE, coinColor = initialsColorPropToColorCode(props), initialsColor = 'white' } = props;

  return {
    root: {
      position: 'relative',
      backgroundColor: coinColor,
      color: initialsColor,
      width: size,
      height: size,
      borderRadius: '50%',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center'
    },
    initials: {
      fontSize: `${size / 2.5}px`
    },
    image: {
      position: 'absolute',
      top: 0,
      left: 0,
      width: size,
      height: size,
      borderRadius: '50%',
      overflow: 'hidden'
    },
    presence: {
      position: 'absolute',
      right: `${getPresenceOffsetForCoinSize(size)}px`,
      bottom: `${getPresenceOffsetForCoinSize(size)}px`
    }
  };
};

function getPresenceOffsetForCoinSize(size: number) {
  // TODO: Tweak this as currently it's more of an approximation.
  return -(size / 15);
}
