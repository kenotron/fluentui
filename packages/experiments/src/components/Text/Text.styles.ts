import { ITextComponent } from './Text.types';

export const TextStyles: ITextComponent['styles'] = props => {
  const { as, className, inline, theme, wrap, variant } = props;
  const { fonts } = theme;
  const variantObject = variant && fonts[variant] ? fonts[variant] : fonts.medium;

  return {
    root: [
      theme.fonts.medium,
      {
        display: inline ? 'inline' : as === 'td' ? 'table-cell' : 'block',
        fontFamily: (variantObject && variantObject.fontFamily) || 'inherit',
        fontSize: (variantObject && variantObject.fontSize) || 'inherit',
        fontWeight: (variantObject && variantObject.fontWeight) || 'inherit',
        color: (variantObject && variantObject.color) || 'inherit',
        mozOsxFontSmoothing: variantObject && variantObject.MozOsxFontSmoothing,
        webkitFontSmoothing: variantObject && variantObject.WebkitFontSmoothing
      },
      !wrap && {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis'
      },
      className
    ]
  };
};
