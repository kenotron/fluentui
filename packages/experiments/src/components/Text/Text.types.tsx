import { IStyle } from '../../Styling';
import { IStatelessComponent, IStyleableComponentProps } from '../../Foundation';
import { IFontStyles } from '../../Styling';

export type ITextComponent = IStatelessComponent<ITextProps, ITextStyles>;

// Styles for the component
export interface ITextStyles {
  /**
   * Style for the root element.
   */
  root: IStyle;
}

// Inputs to the component
export interface ITextProps extends IStyleableComponentProps<ITextProps, ITextStyles> {
  /**
   * Optionally render the component as another component type or primitive.
   */
  as?: string | React.ReactType<ITextProps>;

  /**
   * Optional class name for Text.
   */
  className?: string;

  /**
   * Inline styling.
   */
  style?: React.CSSProperties;

  /**
   * Optional font type for Text.
   */
  variant?: keyof IFontStyles;

  /**
   * Whether the text is displayed as an inline element.
   * Note that inline does not support ellipsis truncation by default.
   */
  inline?: boolean;

  /**
   * Whether the text is wrapped.
   */
  wrap?: boolean;
}
