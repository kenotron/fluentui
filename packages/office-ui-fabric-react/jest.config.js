let { createConfig } = require('../../scripts/tasks/jest-resources');
let path = require('path');

const config = createConfig({
  setupFiles: [path.resolve(path.join(__dirname, 'config', 'tests.js'))],

  moduleNameMapper: {
    // These mappings allow Jest to run snapshot tests against Example files
    'office-ui-fabric-react/lib/(.*)': '<rootDir>/src/$1',
    '@uifabric/example-app-base': '@uifabric/example-app-base/lib-commonjs/utilities/data'
  },

  snapshotSerializers: [require.resolve('@uifabric/jest-serializer-merge-styles')]
});

module.exports = config;
