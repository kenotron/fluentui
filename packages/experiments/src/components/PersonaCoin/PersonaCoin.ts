import { PersonaCoinView } from './PersonaCoin.view';
import { PersonaCoinStyles } from './PersonaCoin.styles';
import { IPersonaCoinProps, IPersonaCoinStyles } from './PersonaCoin.types';
import { createStatelessComponent } from '../../Foundation';

export const PersonaCoin: React.StatelessComponent<IPersonaCoinProps> = createStatelessComponent<IPersonaCoinProps, IPersonaCoinStyles>({
  displayName: 'PersonaCoin',
  view: PersonaCoinView,
  styles: PersonaCoinStyles
});
