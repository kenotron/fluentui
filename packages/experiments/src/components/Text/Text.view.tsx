import * as React from 'react';
import { ITextComponent } from './Text.types';

export const TextView: ITextComponent['view'] = props => {
  const { inline, className, as: RootType = 'span', variant, wrap, classNames, ...rest } = props;

  return <RootType {...rest} className={classNames.root} />;
};
