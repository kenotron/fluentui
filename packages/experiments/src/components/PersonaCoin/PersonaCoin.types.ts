import { ImageLoadState, PersonaPresence, IStyle } from 'office-ui-fabric-react';
import { IStatelessComponent, IStyleableComponentProps } from '../../Foundation';

export type IPersonaCoinComponent = IStatelessComponent<IPersonaCoinProps, IPersonaCoinStyles>;

export type PersonaCoinSize = 10 | 16 | 24 | 28 | 32 | 40 | 48 | 56 | 72 | 100;

// Extending IStyleableComponentProps will automatically add stylable props for you, such as styles and theme.
//    If you don't want these props to be included in your component, just remove this extension.
export interface IPersonaCoinProps extends IStyleableComponentProps<IPersonaCoinProps, IPersonaCoinStyles> {
  /**
   * Whether initials are calculated for phone numbers and number sequences.
   * Example: Set property to true to get initials for project names consisting of numbers only.
   * @defaultvalue false
   */
  allowPhoneInitials?: boolean;

  /**
   * The color that should be used when rendering the coin.
   */
  coinColor?: string;

  /**
   * The user's initials to display in the image area when there is no image.
   * This property can be used instead of `text` to force the initials.
   */
  initials?: string;

  /**
   * This will be used to extract initials from if `initials` is not passed.
   */
  text?: string;

  /**
   * Size of the coin
   * @defaultvalue 48
   */
  size?: PersonaCoinSize;

  /**
   * Url to the image to use, should be a square aspect ratio and big enough to fit in the image area.
   */
  imageUrl?: string;

  /**
   * If true, adds the css class 'is-fadeIn' to the image.
   * @defaultvalue true
   */
  imageShouldFadeIn?: boolean;

  /**
   * If true, the image starts as visible and is hidden on error. Otherwise, the image is hidden until
   * it is successfully loaded. This disables imageShouldFadeIn.
   * @defaultvalue true
   */
  imageShouldStartVisible?: boolean;

  /**
   * Alt text for the image to use. Defaults to an empty string.
   */
  imageAlt?: string;

  /**
   * Optional callback for when loading state of the photo changes
   */
  onPhotoLoadingStateChange?: (newImageLoadState: ImageLoadState) => void;

  /**
   * The color of the user's initials that are displayed.
   * @defaultvalue 'white'
   */
  initialsColor?: string;

  /**
   * Presence of the person to display - will not display presence if undefined.
   * @defaultvalue PersonaPresence.none
   */
  presence?: PersonaPresence;
}

export interface IPersonaCoinStyles {
  /**
   * Styling for the root element.
   */
  root: IStyle;

  /**
   * Styling for the image element
   */
  image: IStyle;

  /**
   * Styling for the initials element
   */
  initials: IStyle;

  /**
   * Styling for the presence element
   */
  presence: IStyle;
}
