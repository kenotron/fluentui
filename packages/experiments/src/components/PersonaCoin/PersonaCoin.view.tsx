import * as React from 'react';
import { Icon } from 'office-ui-fabric-react';
import { Text } from '../../Text';
import { IPersonaCoinComponent } from './PersonaCoin.types';
import { PersonaCoinImage } from './PersonaCoinImage/PersonaCoinImage';
import { PersonaPresence } from 'office-ui-fabric-react/lib/PersonaPresence';
import { getInitials, getRTL } from '../../Utilities';
import { DEFAULT_PERSONA_COIN_SIZE } from './PersonaCoin.styles';

const PersonaCoinInitials: IPersonaCoinComponent['view'] = props => {
  const initials = props.initials || getInitials(props.text, getRTL(), props.allowPhoneInitials);

  if (initials) {
    return <Text className={props.classNames.initials}>{initials}</Text>;
  }

  return <Icon className={props.classNames.initials} iconName="Contact" />;
};

export const PersonaCoinView: IPersonaCoinComponent['view'] = props => {
  const coinSize = props.size || DEFAULT_PERSONA_COIN_SIZE;

  return (
    <div className={props.classNames.root}>
      <PersonaCoinInitials
        coinColor={props.coinColor}
        initials={props.initials}
        text={props.text}
        allowPhoneInitials={props.allowPhoneInitials}
        classNames={props.classNames}
      />
      <PersonaCoinImage
        classNames={props.classNames}
        src={props.imageUrl}
        dimension={coinSize}
        onPhotoLoadingStateChange={props.onPhotoLoadingStateChange}
        imageShouldFadeIn={props.imageShouldFadeIn}
        imageShouldStartVisible={props.imageShouldStartVisible}
        imageAlt={props.imageAlt}
      />
      {props.presence ? <PersonaPresence presence={props.presence} coinSize={coinSize} size={coinSize} styles={props.classNames} /> : null}
    </div>
  );
};
