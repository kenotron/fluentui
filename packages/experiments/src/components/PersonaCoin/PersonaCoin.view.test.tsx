import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Text } from '../../Text';

import { IPersonaCoinStyles } from './PersonaCoin.types';
import { PersonaCoinView } from './PersonaCoin.view';
import { IProcessedStyleSet } from 'office-ui-fabric-react/lib/Styling';
import { PersonaTestImages } from '@uifabric/experiments/lib/common/TestImages';
import { Icon, Image } from 'office-ui-fabric-react';
import { setRTL } from '../../Utilities';

// These tests will ensure that your styles regions have classname representation in snapshot output.
// (Also, classNames is a required prop for views, so we have to supply it for tests.)
const testPersonaCoinClassNames: IProcessedStyleSet<IPersonaCoinStyles> = {
  root: 'test-cn-root',
  image: 'test-cn-image',
  initials: 'test-cn-initials',
  presence: 'test-cn-presence',
  subComponentStyles: {}
};

// Views are just pure functions with no statefulness, which means they can get full code coverage
//    with snapshot tests exercising permutations of the props.
describe('PersonaCoinView', () => {
  it('renders a correct persona', () => {
    const tree = renderer.create(<PersonaCoinView text="James Bond" classNames={testPersonaCoinClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a coin with the initials JB', () => {
    const tree = renderer.create(<PersonaCoinView initials="JB" classNames={testPersonaCoinClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a red coin', () => {
    const tree = renderer.create(<PersonaCoinView initials="JB" classNames={testPersonaCoinClassNames} coinColor="red" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a coin with a contact icon', () => {
    const tree = renderer.create(<PersonaCoinView classNames={testPersonaCoinClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a coin with a contact icon for a Chinese name', () => {
    const tree = renderer.create(<PersonaCoinView text="五号" classNames={testPersonaCoinClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders presence when it is passed', () => {
    const tree = renderer.create(<PersonaCoinView text="五号" presence={4} classNames={testPersonaCoinClassNames} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders presence correctly when a very large coin is rendered', () => {
    const tree = renderer.create(<PersonaCoinView text="五号" presence={4} classNames={testPersonaCoinClassNames} size={100} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the coin with the provided image', () => {
    const tree = renderer
      .create(<PersonaCoinView text="Ellen Grace" imageUrl={PersonaTestImages.personMale} classNames={testPersonaCoinClassNames} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calculates an expected initials in LTR if one was not specified', () => {
    let wrapper = mount(<PersonaCoinView text="Kat Larrson" classNames={testPersonaCoinClassNames} />);
    let result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('KL');
    wrapper.unmount();

    wrapper = mount(<PersonaCoinView text="David Zearing-Goff" classNames={testPersonaCoinClassNames} />);
    result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('DZ');
    wrapper.unmount();

    wrapper = mount(<PersonaCoinView text="4lex 5loo" classNames={testPersonaCoinClassNames} />);
    result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('45');
    wrapper.unmount();

    wrapper = mount(<PersonaCoinView text="Swapnil Vaibhav" classNames={testPersonaCoinClassNames} />);
    result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('SV');
    wrapper.unmount();

    wrapper = mount(<PersonaCoinView text="+1 (555) 6789" classNames={testPersonaCoinClassNames} />);
    result = wrapper.find(Icon);
    expect(result).toHaveLength(1);
    expect(result.props().iconName).toEqual('Contact');
    wrapper.unmount();

    wrapper = mount(<PersonaCoinView text="+1 (555) 6789" allowPhoneInitials={true} classNames={testPersonaCoinClassNames} />);
    result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('16');
    wrapper.unmount();

    wrapper = mount(<PersonaCoinView text="David (The man) Goff" classNames={testPersonaCoinClassNames} />);
    result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('DG');
  });

  it('calculates an expected initials in RTL if one was not specified', () => {
    setRTL(true);

    const wrapper = mount(<PersonaCoinView text="Kat Larrson" classNames={testPersonaCoinClassNames} />);
    const result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('LK');

    setRTL(false);
  });

  it('uses provided initial', () => {
    setRTL(true);
    const wrapper = mount(<PersonaCoinView text="Kat Larrson" initials="AT" classNames={testPersonaCoinClassNames} />);
    const result = wrapper.find(Text);
    expect(result).toHaveLength(1);
    expect(result.props().children).toEqual('AT');

    setRTL(false);
  });

  describe('image', () => {
    const testImage1x1 =
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAAC0lEQVQImWP4DwQACfsD/eNV8pwAAAAASUVORK5CYII=';

    it('renders empty alt text by default', () => {
      const wrapper = mount(<PersonaCoinView text="Kat Larrson" imageUrl={testImage1x1} classNames={testPersonaCoinClassNames} />);
      const image = wrapper.find(Image);

      expect(image.props().alt).toEqual('');
    });

    it('renders its given alt text', () => {
      const wrapper = mount(
        <PersonaCoinView text="Kat Larrson" imageUrl={testImage1x1} imageAlt="ALT TEXT" classNames={testPersonaCoinClassNames} />
      );
      const image = wrapper.find(Image);

      expect(image.props().alt).toEqual('ALT TEXT');
    });
  });
});
