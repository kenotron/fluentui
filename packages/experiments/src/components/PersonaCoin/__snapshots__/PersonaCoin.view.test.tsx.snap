// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonaCoinView renders a coin with a contact icon 1`] = `
<div
  className="test-cn-root"
>
  <i
    aria-hidden={true}
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          display: inline-block;
          font-family: "FabricMDL2Icons-0";
          font-style: normal;
          font-weight: normal;
          speak: none;
        }
    data-icon-name="Contact"
    role="presentation"
  >
    
  </i>
</div>
`;

exports[`PersonaCoinView renders a coin with a contact icon for a Chinese name 1`] = `
<div
  className="test-cn-root"
>
  <i
    aria-hidden={true}
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          display: inline-block;
          font-family: "FabricMDL2Icons-0";
          font-style: normal;
          font-weight: normal;
          speak: none;
        }
    data-icon-name="Contact"
    role="presentation"
  >
    
  </i>
</div>
`;

exports[`PersonaCoinView renders a coin with the initials JB 1`] = `
<div
  className="test-cn-root"
>
  <span
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          overflow: hidden;
          text-overflow: ellipsis;
          webkit-font-smoothing: antialiased;
          white-space: nowrap;
        }
  >
    JB
  </span>
</div>
`;

exports[`PersonaCoinView renders a correct persona 1`] = `
<div
  className="test-cn-root"
>
  <span
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          overflow: hidden;
          text-overflow: ellipsis;
          webkit-font-smoothing: antialiased;
          white-space: nowrap;
        }
  >
    JB
  </span>
</div>
`;

exports[`PersonaCoinView renders a red coin 1`] = `
<div
  className="test-cn-root"
>
  <span
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          overflow: hidden;
          text-overflow: ellipsis;
          webkit-font-smoothing: antialiased;
          white-space: nowrap;
        }
  >
    JB
  </span>
</div>
`;

exports[`PersonaCoinView renders presence correctly when a very large coin is rendered 1`] = `
<div
  className="test-cn-root"
>
  <i
    aria-hidden={true}
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          display: inline-block;
          font-family: "FabricMDL2Icons-0";
          font-style: normal;
          font-weight: normal;
          speak: none;
        }
    data-icon-name="Contact"
    role="presentation"
  >
    
  </i>
  <div
    className=
        ms-Persona-presence
        test-cn-presence
        {
          -ms-high-contrast-adjust: none;
          background-clip: content-box;
          background-color: #E81123;
          border-radius: 50%;
          border: 2px solid #ffffff;
          bottom: -2px;
          box-sizing: content-box;
          height: 12px;
          position: absolute;
          right: -2px;
          text-align: center;
          top: auto;
          width: 12px;
        }
        @media screen and (-ms-high-contrast: active){& {
          background-color: WindowText;
          border-color: Window;
        }
    style={
      Object {
        "height": "33.333333333333336px",
        "width": "33.333333333333336px",
      }
    }
  >
    <i
      aria-hidden={true}
      className=
          ms-Persona-presenceIcon
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #ffffff;
            display: inline-block;
            font-family: "FabricMDL2Icons";
            font-size: 6px;
            font-style: normal;
            font-weight: normal;
            line-height: 12px;
            speak: none;
            vertical-align: top;
          }
          @media screen and (-ms-high-contrast: active){& {
            color: Window;
          }
      data-icon-name="SkypeMinus"
      role="presentation"
      style={
        Object {
          "fontSize": "16.666666666666668px",
          "lineHeight": "33.333333333333336px",
        }
      }
    >
      
    </i>
  </div>
</div>
`;

exports[`PersonaCoinView renders presence when it is passed 1`] = `
<div
  className="test-cn-root"
>
  <i
    aria-hidden={true}
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          display: inline-block;
          font-family: "FabricMDL2Icons-0";
          font-style: normal;
          font-weight: normal;
          speak: none;
        }
    data-icon-name="Contact"
    role="presentation"
  >
    
  </i>
  <div
    className=
        ms-Persona-presence
        test-cn-presence
        {
          -ms-high-contrast-adjust: none;
          background-clip: content-box;
          background-color: #E81123;
          border-radius: 50%;
          border: 2px solid #ffffff;
          bottom: -2px;
          box-sizing: content-box;
          height: 12px;
          position: absolute;
          right: -2px;
          text-align: center;
          top: auto;
          width: 12px;
        }
        @media screen and (-ms-high-contrast: active){& {
          background-color: WindowText;
          border-color: Window;
        }
    style={
      Object {
        "height": "16px",
        "width": "16px",
      }
    }
  >
    <i
      aria-hidden={true}
      className=
          ms-Persona-presenceIcon
          {
            -moz-osx-font-smoothing: grayscale;
            -webkit-font-smoothing: antialiased;
            color: #ffffff;
            display: inline-block;
            font-family: "FabricMDL2Icons";
            font-size: 6px;
            font-style: normal;
            font-weight: normal;
            line-height: 12px;
            speak: none;
            vertical-align: top;
          }
          @media screen and (-ms-high-contrast: active){& {
            color: Window;
          }
      data-icon-name="SkypeMinus"
      role="presentation"
      style={
        Object {
          "fontSize": "8px",
          "lineHeight": "16px",
        }
      }
    >
      
    </i>
  </div>
</div>
`;

exports[`PersonaCoinView renders the coin with the provided image 1`] = `
<div
  className="test-cn-root"
>
  <span
    className=
        test-cn-initials
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          color: inherit;
          display: block;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          moz-osx-font-smoothing: grayscale;
          overflow: hidden;
          text-overflow: ellipsis;
          webkit-font-smoothing: antialiased;
          white-space: nowrap;
        }
  >
    EG
  </span>
  <div
    className=
        ms-Image
        test-cn-image
        {
          -moz-osx-font-smoothing: grayscale;
          -webkit-font-smoothing: antialiased;
          border-radius: 50%;
          font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          font-size: 14px;
          font-weight: 400;
          overflow: hidden;
          position: relative;
        }
    style={
      Object {
        "height": 48,
        "width": 48,
      }
    }
  >
    <img
      alt=""
      className=
          ms-Image-image
          ms-Image-image--cover
          ms-Image-image--portrait
          is-notLoaded
          is-fadeIn
          {
            display: block;
            height: auto;
            left: 50% /* @noflip */;
            opacity: 0;
            position: absolute;
            top: 50%;
            transform: translate(-50%,-50%);
            width: 100%;
          }
      onError={[Function]}
      onLoad={[Function]}
      src="https://static2.sharepointonline.com/files/fabric/office-ui-fabric-react-assets//persona-male.png"
    />
  </div>
</div>
`;
